src/Util/MigrationRollbackBatch.php

```diff
@@ -238,11 +239,14 @@ public function processFromQueue(array $item) {
       $id_map->delete($item['source']);
     }
     else {
-      $this->getEventDispatcher()
-        ->dispatch(new MigrateRowDeleteEvent($this->migration, $item['destination']), MigrateEvents::PRE_ROW_DELETE);
-      $destination->rollback($item['destination']);
-      $this->getEventDispatcher()
-        ->dispatch(new MigrateRowDeleteEvent($this->migration, $item['destination']), MigrateEvents::POST_ROW_DELETE);
+      $map_row = $id_map->getRowByDestination($item['destination']);
+      if (!isset($map_row['rollback_action']) || $map_row['rollback_action'] === MigrateIdMapInterface::ROLLBACK_DELETE) {
```

Comment:
Strict type checking causes rollbacks to fail and mirrors core's implementation: https://git.drupalcode.org/project/drupal/-/blob/10.3.x/core/modules/migrate/src/MigrateExecutable.php?ref_type=heads#L339

```suggestion
      if (!isset($map_row['rollback_action']) || $map_row['rollback_action'] == MigrateIdMapInterface::ROLLBACK_DELETE) {
```